charts/kube-prometheus-stack/values.yaml

```diff
@@ -3251,6 +3251,25 @@ prometheus:
       #   hosts:
       #     - prometheus.example.com
 
+  gateway:
```

Comment:
Do you have other examples of this kind of flag upstream? Because gateway is used for a lot of other things right now liké Mimir gateways so maybe other upstream projects use better names liké gatewayapi?  It could also be they use `gateway`😅

Reply:
No. I found nothing via GitHub Code search. Bitnami and Grafana did not have Gateway API yet, but I'm open for other names here.

But gatewayApi sounds more specific.

Reply:
I would propose to skip the gateway level and use the httpRoute level only, this makes sense with gateway api terminology where the gateway resource is managed centrally by the cluster operator and the application developer only deals with routes. If the extra level is needed because you envision that tlsRoute etc might also at some point be supported I would propose the following structure:

```yaml
routes:
  http:
  tls:
  grpc:

etc etc
```

Reply:
Still prefix the values with `gatewayAPI` - who knows what comes? 

Currently, only httpRoute is implemented and there are no plans to provide more

Reply:
Sorry for not getting back to you. I still prefer `routes` as the terminology as this is how the concept is called in gateway api: https://gateway-api.sigs.k8s.io/concepts/security-model/#resources

But no hard feelings over calling it something else, I understand that since it's specific to gateway api you may want to scope the name better so there are no misunderstandings.

Reply:
I would be fine with routes as well but we need to be careful about openshift users

Reply:
That is a good point, could those perhaps be merged or are they not similar enough? Conceptually it would be nice if openshift routes could simply be treated as a different route type instead of it's own separate construct but I am not familiar enough with openshift to say.

I did a quick search and it seems like the authors of the helm chart library `common` at least have decided on `route` for now:

Both in truecharts and in the upstream version by bjw-s:

https://github.com/truecharts/public/blob/master/charts/library/common/values.yaml#L684
https://github.com/bjw-s/helm-charts/blob/main/charts/library/common/values.yaml#L653

And I also find it in other places:

https://github.com/harshaputtaswamy/oom/blob/master/kubernetes/common/common/templates/_ingress.tpl#L332

Reply:
Then maybe we should follow along yes with that :) @jkroepke what do you think?

Reply:
I like some ideas and I did not had v1alpha2 in scope.

@QuentinBisson and @rouke-broersma please take another look of that example.


```
helm template . -f ci/05-ingress-and-gateway-routes-values.yaml -s templates/prometheus/route.yaml
---
# Source: kube-prometheus-stack/templates/prometheus/route.yaml
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: release-name-kube-promethe-prometheus
  namespace: default
  labels:
    app: kube-prometheus-stack-prometheus
    
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "65.5.1"
    app.kubernetes.io/part-of: kube-prometheus-stack
    chart: kube-prometheus-stack-65.5.1
    release: "release-name"
    heritage: "Helm"
spec:
  hostnames:
    - '*.test.com'
  rules:
    - backendRefs:
        - name: release-name-kube-promethe-prometheus
          port: 9090
      matches:
        - path:
            type: PathPrefix
            value: /

```

Reply:
I am not a fan of the common chart and do not use them myself! It was just an example of what I found in the wild.

I like the current proposal, and I think it exposes sufficient fields for an initial version. Other fields can be added as needed when we get more experience with gateway api. 

I am not sure route kind is even necessary for prometheus, I think only http route can be valid? That said it doesn't hurt either and could provide easy extension later if a different route type becomes relevant to prometheus so I'm also fine with leaving it in.